src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -18,148 +18,217 @@ package nl.knaw.dans.easy.deposit.servlets
+    (for {
+      uuid <- getUUID
+      managedIS <- getRequestBodyAsManagedInputStream
+      datasetMetadata <- managedIS.apply(is => DatasetMetadata(is))
+      _ <- app.writeDataMetadataToDeposit(datasetMetadata, user.id, uuid)
+    } yield NoContent()
+      ).getOrRecoverResponse(respond)
   }
   get("/:uuid/state") {
-    forDeposit(app.getDepositState)
-      .map(depositState => Ok(body = toJson(depositState)))
-      .getOrRecoverResponse(respond)
+    (for {
+      uuid <- getUUID
+      depositState <- app.getDepositState(user.id, uuid)
+    } yield Ok(body = toJson(depositState))
+      ).getOrRecoverResponse(respond)
   }
   put("/:uuid/state") {
-    {
-      for {
-        managedIS <- getRequestBodyAsManagedInputStream
-        stateInfo <- managedIS.apply(is => StateInfo(is))
-        _ <- forDeposit(app.setDepositState(stateInfo))
-      } yield NoContent()
-    }.getOrRecoverResponse(respond)
+    (for {
+      uuid <- getUUID
+      managedIS <- getRequestBodyAsManagedInputStream
+      stateInfo <- managedIS.apply(is => StateInfo(is))
+      _ <- app.setDepositState(stateInfo, user.id, uuid)
+    } yield NoContent()
+      ).getOrRecoverResponse(respond)
   }
 
   delete("/:uuid") {
-    forDeposit(app.deleteDeposit)
-      .map(_ => NoContent())
-      .getOrRecoverResponse(respond)
+    (for {
+      uuid <- getUUID
+      _ <- app.deleteDeposit(user.id, uuid)
+    } yield NoContent()
+      ).getOrRecoverResponse(respond)
   }
   get("/:uuid/file/*") { //dir and file
-    forPath(app.getDepositFiles)
-      .map(depositFiles => Ok(body = toJson(depositFiles)))
-      .getOrRecoverResponse(respond)
-  }
-  post("/:uuid/file/*") { upload } //dir
-  put("/:uuid/file/*") { upload } //file
-  private def upload = {
-    for {
-      managedIS <- getRequestBodyAsManagedInputStream
-      newFileWasCreated <- managedIS.apply(is => forPath(app.writeDepositFile(is)))
-    } yield if (newFileWasCreated)
-              Created(headers = Map("Location" -> request.uri.toASCIIString))
-            else Ok()
-  }.getOrRecoverResponse(respond)
-
-  delete("/:uuid/file/*") { //dir and file
-    forPath(app.deleteDepositFile)
-      .map(_ => NoContent())
-      .getOrRecoverResponse(respond)
-  }
-
-  private def forUser[T](callback: String => Try[T]
-                        ): Try[T] = {
-    // shortest callBack signature
-    // the signatures reflect the parameters in the route patterns and authenticated user
-    Try(callback(user.id)).flatten // catch throws that slipped through
+    (for {
+      uuid <- getUUID
+      path <- getPath
+      depositFiles <- app.getDepositFiles(user.id, uuid, path)
+    } yield Ok(body = toJson(depositFiles))
+      ).getOrRecoverResponse(respond)
   }
 
-  private def forDeposit[T](callback: (String, UUID) => Try[T]
-                           ): Try[T] = {
-    for {
+  post("/:uuid/file/*") { //file
+    val zip = "application/zip"
+    val bin = "application/octet-stream"
```

Comment:
Content-type is more complex: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17